### PR TITLE
Fix `get_event_log` XML-RPC endpoint

### LINKED ISSUE #3708: [Backport] Fix `get_event_log` XML-RPC endpoint
### Original feature issue

- Issue: #3164 #3263
- PR: #3244 #3264

### Target release

- [x] release33
- [ ] release32
- [ ] release30

### Reason

Enablement of the Golang CLI for Cobbler 3.3.x release series.

### PR DESCRIPTION
**Pull request by SchoolGuy**
## Linked Items

Fixes #3708

## Description

See title.

```
e7e97e005e69:/code # curl -XPOST -d '<?xml version="1.0" encoding="UTF-8"?><methodCall><methodName>get_event_log</methodName><params><param><value><string>2024-06-07_132819_Create bootable bootloader images_866a8db02c824ee7aa46037f8840180d</string></value></param></params></methodCall>' http://localhost:25151
<?xml version='1.0'?>
<methodResponse>
<params>
<param>
<value><string>[2024-06-07_132819_Create bootable bootloader images_866a8db02c824ee7aa46037f8840180d] 2024-06-07 13:28:19,595 - INFO | start_task(mkloaders); event_id(2024-06-07_132819_Create bootable bootloader images_866a8db02c824ee7aa46037f8840180d)
[2024-06-07_132819_Create bootable bootloader images_866a8db02c824ee7aa46037f8840180d] 2024-06-07 13:28:19,601 - INFO | syslinux version 4 detected! Skip making symlink of "ldlinux.c32" file!
[2024-06-07_132819_Create bootable bootloader images_866a8db02c824ee7aa46037f8840180d] 2024-06-07 13:28:19,601 - INFO | GRUB2 modules directory for arch "aarch64" did no exist. Skipping GRUB2 creation
[2024-06-07_132819_Create bootable bootloader images_866a8db02c824ee7aa46037f8840180d] 2024-06-07 13:28:19,601 - INFO | GRUB2 modules directory for arch "arm" did no exist. Skipping GRUB2 creation
[2024-06-07_132819_Create bootable bootloader images_866a8db02c824ee7aa46037f8840180d] 2024-06-07 13:28:19,677 - INFO | Successfully built bootloader for arch "arm64-efi"!
[2024-06-07_132819_Create bootable bootloader images_866a8db02c824ee7aa46037f8840180d] 2024-06-07 13:28:19,678 - INFO | GRUB2 modules directory for arch "i386-efi" did no exist. Skipping GRUB2 creation
[2024-06-07_132819_Create bootable bootloader images_866a8db02c824ee7aa46037f8840180d] 2024-06-07 13:28:19,987 - INFO | Successfully built bootloader for arch "i386-pc-pxe"!
[2024-06-07_132819_Create bootable bootloader images_866a8db02c824ee7aa46037f8840180d] 2024-06-07 13:28:19,987 - INFO | GRUB2 modules directory for arch "i686" did no exist. Skipping GRUB2 creation
[2024-06-07_132819_Create bootable bootloader images_866a8db02c824ee7aa46037f8840180d] 2024-06-07 13:28:19,987 - INFO | GRUB2 modules directory for arch "IA64" did no exist. Skipping GRUB2 creation
[2024-06-07_132819_Create bootable bootloader images_866a8db02c824ee7aa46037f8840180d] 2024-06-07 13:28:19,987 - INFO | GRUB2 modules directory for arch "powerpc-ieee1275" did no exist. Skipping GRUB2 creation
[2024-06-07_132819_Create bootable bootloader images_866a8db02c824ee7aa46037f8840180d] 2024-06-07 13:28:20,058 - INFO | Successfully built bootloader for arch "x86_64-efi"!
[2024-06-07_132819_Create bootable bootloader images_866a8db02c824ee7aa46037f8840180d] 2024-06-07 13:28:20,058 - INFO | ### TASK COMPLETE ###
</string></value>
</param>
</params>
</methodResponse>
```

## Behaviour changes

Old: Task logs were missing from `/var/log/cobbler/tasks`, subsequently the XML-RPC endpoint was not returning content.

New: Cobbler now correctly adds task logs again and write specific task logs. The endpoint works again (see description).

## Category

This is related to a:

- [x] Bugfix
- [ ] Feature
- [ ] Packaging
- [ ] Docs
- [ ] Code Quality
- [ ] Refactoring
- [ ] Miscellaneous

## Tests

- [ ] Unit-Tests were created
- [ ] System-Tests were created
- [ ] Code is already covered by Unit-Tests
- [ ] Code is already covered by System-Tests
- [x] No tests required

### CODE REVIEW

**Comment by SchoolGuy** [comment]
Close and reopen for CI retrigger.

**Comment by SchoolGuy** [comment]
This needs to take the status if #3682 into account as they both touch `CobblerThread`.

File: cobbler/remote.py [hunk 1]
```diff
@@ -440,37 +378,29 @@ def get_event_log(self, event_id: str) -> str:
...
         :param name: The name of the event.
         """
-        event_id = self.__generate_event_id("event")
-        event_id = str(event_id)
-        self.events[event_id] = [float(time.time()), str(name), EVENT_INFO, []]
+        new_event = utils.CobblerEvent(name=name, statetime=time.time())
+        self.events[new_event.event_id] = new_event
+        return new_event
 
     def __start_task(self, thr_obj_fn, token: str, role_name: str, name: str, args: dict, on_done=None):
```
Comment (m-czernek) [inline cobbler/remote.py:405]: Would it make sense to rename `args` to `options` here? Seems like that's the naming used throughout the codebase for this (both by code that call as well as use this method)

**Review by m-czernek — APPROVED** [review]
+1, leaving a minor comment. Retested, works as expected.

**Pull request merged**